yt/geometry/particle_geometry_handler.py

```diff
@@ -128,13 +129,14 @@ def _initialize_index(self):
         else:
             fname = ds.index_filename
 
+        dont_load = dont_cache and not hasattr(ds, 'index_filename')
         try:
             rflag = self.regions.load_bitmasks(fname)
             rflag = self.regions.check_bitmasks()
             self._initialize_frontend_specific()
-            if rflag == 0:
+            if dont_load or rflag == 0:
                 raise OSError
```

Comment:
If I follow this: in the case that we don't want to load, `dont_load = True` we still try and load the bitmasks, but then we raise an `OSError` which triggers the exception and thus, we then initialize from scratch? 

If so, I guess this is ok, because loading is fast.

Reply:
That was the idea, yeah, but I think maybe it's not the best idea anymore.  It might need to be worked through a bit more carefully.

Reply:
I guess you could put the `if dont_load: raise OSError` straight after the `try` and then at the end of the `try` have the `if flag == 0: raise OSError` - otherwise `yt` outputs:
`Loading particle index...`
`Generating particle index...`

Reply:
good point, will do this